File: Aufgabe_1/Aufgabe/A1-WiSe1920Counter/base_counter.rb
require_relative 'counter'
class BaseCounter

  include Counter

  def initialize(start=0)
    @start=start
    @count=start
  end

  def start()
    return @start
  end

  def inc()
    @count+=1
    return self
  end

  def tally()
    return @count
  end

  def reset()
    @count=@start
  end

end

bc = BaseCounter.new(4)
bc.inc().inc().inc()
puts bc
bc.reset()
bc.inc().inc()
puts bc

File: A1/src/a11/BaseCounter.java
package a11;

public  class BaseCounter extends Counter{
    private int start;
    protected int count;

    public BaseCounter(int start){
        this.start = start;
        this.count = start;
    }

    public BaseCounter(){
        this(0);
    }

    @Override
    public Counter inc() {
        this.count++;
        return this;
    }

    @Override
    public int start() {
        return this.start;
    }

    @Override
    public int tally() {
        return this.count;
    }

    @Override
    public void reset(){
    this.count = this.start;
    }

    public static void main(String[] args) {
        BaseCounter bc = new BaseCounter(4);
        bc.inc();
        bc.inc();bc.inc();
        System.out.println(bc);
        bc.reset();
        bc.inc();bc.inc();
        System.out.println(bc);
    }

}

